spec/factories/events.rb

```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :event do
+    speaker { nil }
```

Comment:
it should be 
`association :speaker, factory: :user` 
innstead of
`speaker { nil }`
It associates an Event factory with a User factory and provides value for the field that is supposed to be `null: false`

Reply:
I don't think Tracy made any tests. I think these are auto generated when you generate a model. We can write tests for this later.